domain/src/main/kotlin/no/nav/su/se/bakover/domain/beregning/beregning/Månedsberegning.kt

```diff
@@ -0,0 +1,61 @@
+package no.nav.su.se.bakover.domain.beregning.beregning
+
+import no.nav.su.se.bakover.common.Tidspunkt
+import no.nav.su.se.bakover.common.limitedUpwardsTo
+import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.common.periode.PeriodisertInformasjon
+import no.nav.su.se.bakover.common.positiveOrZero
+import no.nav.su.se.bakover.domain.Grunnbeløp
+import no.nav.su.se.bakover.domain.beregning.Sats
+import no.nav.su.se.bakover.domain.beregning.fradrag.IFradrag
+import java.util.UUID
+
+interface IMånedsberegning : PeriodisertInformasjon {
+    fun id(): UUID
+    fun opprettet(): Tidspunkt
+    fun sum(): Double
+    fun fradrag(): Double
+    fun grunnbeløp(): Int
+    fun sats(): Sats
+}
+
+abstract class AbstractMånedsberegning : IMånedsberegning {
+    private val id by lazy { UUID.randomUUID() }
+    private val opprettet by lazy { Tidspunkt.now() }
+    override fun id(): UUID = id
+    override fun opprettet() = opprettet
+}
+
+internal data class Månedsberegning(
+    private val periode: Periode,
+    private val sats: Sats,
+    internal val fradrag: List<IFradrag>
+) : AbstractMånedsberegning() {
+    init {
+        require(fradrag.all { it.periode() == periode }) { "Fradrag må være gjeldende for aktuell måned" }
+        require(periode.antallMåneder() == 1) { "Månedsberegning kan kun utføres for en enkelt måned" }
+    }
+
+    override fun sum() = (periodiserSats() - fradrag())
+        .positiveOrZero()
+
+    override fun fradrag() = fradrag
+        .sumByDouble { it.månedsbeløp() }
+        .limitedUpwardsTo(periodiserSats())
+
+    override fun grunnbeløp(): Int = Grunnbeløp.`1G`.fraDato(periode.fraOgMed()).toInt()
+    override fun sats(): Sats = sats
+
+    private fun periodiserSats() = sats.periodiser(periode).getValue(periode)
+
+    override fun periode(): Periode = periode
+}
+
+data class MånedsberegningDbWrapper(
```

Comment:
Jeg syns navngivingen her var litt vanskelig. Det høres ut som at det er en wrapper rundt databasefunksjonalitet for månedsberegninger, men av implementasjonen ser det ut som at det bare er en type for å også ha `id` og `opprettet`. Stemmer det?
I så fall, kunne vi ha funnet noe litt tydeligere? Halvteit forslag: `MånedsberegningWithDbFields`.

Reply:
Det stemmer at dette er en klasse hvis formål er å "frikoble" domeneobjektet vårt fra det å holde på db-feltene id og opprettet. Er lei av bloated konstruktører og å styre med ID-generatorer og klokker for å få testet domenelogikk. Må gjøre noen endringer her uansett, har tenkt i retninger av å kalle det for noe sånt som "PersistertXXX", hva tror du om det?